Userland/Utilities/nc.cpp

```diff
@@ -61,7 +61,11 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     args_parser.add_option(udp_mode, "UDP mode", "udp", 'u');
     args_parser.add_option(should_close, "Close connection after reading stdin to the end", nullptr, 'N');
     args_parser.add_option(maximum_tcp_receive_buffer_size_input, "Set maximum tcp receive buffer size", "length", 'I', nullptr);
-    args_parser.add_positional_argument(target, "Address to listen on, or the address or hostname to connect to", "target");
+    args_parser.parse(arguments, Core::ArgsParser::FailureBehavior::Ignore);
+    if (should_listen)
+        target = "127.0.0.1";
```

Comment:
This won't work, as the arguments aren't parsed until we call `args_parser.parse(arguments);` below

Reply:
It does work because I call args_parser.parse twice.  The first call is to determine if “-l” is specified.  If it is, I make the target paramenter optional then call parse again.  It’s not particularly elegant, but I don't think ArgsParser has a built-in way to do this.

Reply:
How does --help work in that case? It would only ever see half the arguments?

Reply:
Good catch, it prints the help message twice.  I can fix this if I add a getter for ArgsParse.m_show_help

Reply:
I still don't think this is a good solution, now it's impossible for the later command line arguments to ever get emitted in the help/usage, since we will always bail out before we add them?